app/code/RefreshMarkdownTask.php

```diff
@@ -106,6 +106,10 @@ private function cloneRepository(array $repository)
      */
     private function cleanRepository(array $repository)
     {
+	
+	// if the dev=1 flag is used when invoking RefreshMarkdownTask the git repos are kept.
+	if( isset($_REQUEST['dev']) && $_REQUEST['dev'] ) return;
```

Comment:
Probably needs to change the git clone command too to ensure you get the full history not just the shallow clone.